pennylane/ops/op_math/condition.py

```diff
@@ -99,7 +99,7 @@ class Conditional(SymbolicOp, Operation):
             can be useful for some applications where the instance has to be identified
     """
 
-    def __init__(self, expr, then_op: type[Operation], id=None):
+    def __init__(self, expr, then_op: Operation, id=None):
```

Comment:
Should this be `Operator` instead? Technically you can use `Prod` as an operation, but `Operation` is not one of its bases.

Reply:
I feel like we need a better abstraction, but for now I'm still more inclined to leave this as is, since it's always been labeled as an `Operation`.